src/test/java/lambda/part2/exercise/Exercise2.java

```diff
@@ -0,0 +1,100 @@
+package lambda.part2.exercise;
+
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.function.Predicate;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+public class Exercise2 {
+
+    @Test
+    public void personHasNotEmptyLastNameAndFirstName() {
+        // TODO предикат Person -> boolean, проверяющий что имя и фамилия человека не пусты
+        Predicate<Person> validate = (person) -> !person.getFirstName().equals("") && !person.getLastName().equals("");
+
+        assertTrue(validate.test(new Person("Алексей", "Доренко", 40)));
+        assertFalse(validate.test(new Person("Николай", "", 30)));
+        assertFalse(validate.test(new Person("", "Мельников", 20)));
+    }
+
+    // TODO метод (Person -> boolean) -> (Person -> boolean)
+    // TODO - возвращает новый предикат, являющийся отрицанием исходного
+    // TODO - при реализации использовать логический оператор !
+    private Predicate<Person> negateUsingLogicalOperator(Predicate<Person> predicate){
+        return  (Person) -> !predicate.test(Person);
+    }
+
+    // TODO метод (Person -> boolean, Person -> boolean) -> (Person -> boolean)
+    // TODO - возвращает новый предикат, объединяющий исходные с помощью операции "AND"
+    // TODO - при реализации использовать логический оператор &&
+    private Predicate<Person> andUsingLogicalOperator(Predicate<Person> left, Predicate<Person> right) {
+        return (Person) -> left.test(Person) && right.test(Person);
+    }
+
+    @Test
+    public void personHasNotEmptyLastNameAndFirstNameUsingLogicalOperators() {
+        Predicate<Person> personHasEmptyFirstName = (person) -> person.getFirstName().equals("");
+        Predicate<Person> personHasEmptyLastName = (person) -> person.getLastName().equals("");
+
+        Predicate<Person> personHasNotEmptyFirstName = (person) -> !person.getFirstName().equals("");
```

Comment:
Sure, you can use your expression here, but here is meant ```negateUsingLogicalOperator(personHasEmptyFirstName)```. 

The same can be said about similar cases:
- ```negateUsingLogicalOperator()``` here and at line 44.
- ```andUsingLogicalOperator()``` at line 46.
- ```negate()``` (from line 57) at lines 73-74.
- ```and()``` (from line 64) at lines 76.

Reply:
It is in the name of the test UsingLogicalOperators

Reply:
But there are methods with these names like ```andUsingLogicalOperator()```, ```negateUsingLogicalOperator()```. 
You already have implemented them as required.

Anyway, I'm sure that we need to solve this task by that way. 
On the other hand, third person can help us to solve the question.